containertools/cmd/nbwatch/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	defer w.Close()
+
+	w.Add("/content/src")
+
+	go watchLoop(w)
+	<-make(chan struct{}) // Block forever
+
+	return nil
+}
+
+func watchLoop(w *fsnotify.Watcher) {
+	i := int64(0)
+	for {
+		select {
+		// Read from Errors.
+		case err, ok := <-w.Errors:
+			if !ok { // Channel was closed (i.e. Watcher.Close() was called).
+				return
+			}
+			klog.Error(err)
+		// Read from Events.
+		case e, ok := <-w.Events:
+			if !ok { // Channel was closed (i.e. Watcher.Close() was called).
+				return
```

Comment:
should we log the event or will `e` be null?

Reply:
In this if block I think `e` is its empty value.